### PR TITLE
bridge: T3715: Standby port supporting

### PR DESCRIPTION
**Pull request by jack9603301**
## Change Summary

Standby port supporting

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3715

## Component(s) name

bridge

## Proposed changes

Standby port supporting

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: python/vyos/ifconfig/bridge.py [hunk 1]
```diff
@@ -365,6 +365,21 @@ def update(self, config):
                     if native_vlan_id:
                         cmd = f'bridge vlan add dev {interface} vid {native_vlan_id} pvid untagged master'
                         self._cmd(cmd)
+            
+            # When the bridge port is initialized, set the standby port now,
+            # This reports an error if the bridge did not complete the member
+            # port addition before setting the standby port
+            for interface, interface_config in member_intf.items():
```
Comment (c-po) [inline python/vyos/ifconfig/bridge.py:372]: Duplicated code in interface.py and bridge.py

Comment (jack9603301) [reply python/vyos/ifconfig/bridge.py:372]: done

File: src/conf_mode/interfaces-bridge.py [hunk 2]
```diff
@@ -123,6 +130,14 @@ def verify(bridge):
             if 'is_bond_member' in interface_config:
                 tmp = next(iter(interface_config['is_bond_member']))
                 raise ConfigError(error_msg + f'it is already a member of bond "{tmp}"!')
+            
+            if 'backup_port' in interface_config and 'backup_port_exist' not in interface_config:
+                tmp = interface_config['backup_port']
+                raise ConfigError(error_msg + f'Device {tmp} is not a bridge member and cannot be used as an alternate port!')
+            
+            if 'backup_port' in interface_config and 'backup_port_exist' in interface_config:
+                if not re.search('^((eth|lan)[0-9]+|(eno|ens|enp|enx).+)$', interface):
```
Comment (c-po) [inline src/conf_mode/interfaces-bridge.py]: This is redundant code - we have an interface name validator which should be used from the XML level. This prevents invalid input to pass down into the Python script.

The validator is named `interface-name` and also supports VLAN interfaces - those can not be used for a bridge backup port, but the validator can be adjusted to validate with and without vlans.

**Review by c-po — CHANGES_REQUESTED** [review]

File: src/conf_mode/interfaces-bridge.py (hunk 2 above)
Comment (jack9603301) [reply src/conf_mode/interfaces-bridge.py]: done
But now I want to limit the use to the default Ethernet network card in the system, so I add a verifier based on this verifier.

Comment (c-po) [reply src/conf_mode/interfaces-bridge.py]: Please rename the validator to `interface-name-eth`

Comment (jack9603301) [reply src/conf_mode/interfaces-bridge.py]: done

**Comment by jack9603301** [comment]
@c-po Can you review the following

**Comment by c-po** [comment]
Hi Jack - what is the purpose of this feature? I'd rather stick with a proper STP solution.

**Comment by jack9603301** [comment]
@c-po Generally, Linux Bridge supports a standby port. According to the man manual, when a specific primary port stops working, traffic is automatically redirected to the standby port

Linux Bridge supports this, and if it does, it might be useful in some cases.

**Comment by c-po** [comment]
Closing this as I unfortunately still did not got the use-case that I can solve using STP.

**Comment by jack9603301** [comment]
> Closing this as I unfortunately still did not got the use-case that I can solve using STP.

Sorry I haven't been following this for a long time, I don't quite understand what you mean

**Pull request closed without merging**